cucumber-messages/ruby/cucumber-messages.gemspec

```diff
@@ -29,8 +29,15 @@ Gem::Specification.new do |s|
   # pick the appropriate one in their bundle.
   #
   # Users of JRuby would probably install 3.2.0, while users of MRI would use
-  # 3.6.0.
-  s.add_dependency 'google-protobuf', '~> 3.2'
+  # a version either around 3.2 (If they are on an old Ruby), or 3.7 if they
+  # are running on Ruby 2.6. This is due to some extensions failing to compile
+  # which has only been recently fixed and won't be backported
+  #
+  if RbConfig::CONFIG["MINOR"] == "6"
+    s.add_dependency('google-protobuf', '~> 3.7')
+  else
+    s.add_dependency('google-protobuf', ['>= 3.2', '< 3.6'])
```

Comment:
I remember this kind of logic not being included in the gem once it's built. Does this work now?

Reply:
I ran this locally on jruby 2.2 and 2.6 and it seemed to work.

That's not to say that's a rigorous test. Just what I'm capable of doing locally. I think we should merge this as is. And if it breaks things we can always revert it.

Reply:
I'm worried about what happens when a gem is build and pushed.

Can you try `gem build` please and see if the resulting gem file contains this logic or just picks one of the two `add_dependency` lines?

Reply:
I've built it, but I cannot inspect it as it's all machine code. Is there a way I can directly test if this logic is here? Might need help to maybe add it to a gemfile or something

Reply:
You can extract the data using, e.g.:
```
tar xvf  cucumber-messages-2.1.1.gem 
```

This will give you
```
metadata.gz
data.tar.gz
checksums.yaml.gz
```

Then you can look at the metadata:
```
gunzip metadata
more metadata
```

Reply:
Nope, it's not pulling through the conditionals. Urm I'll revert the bit in the gemfile. See if that fixes things. After that I'm not sure.

Although (cynically), I'd rather get it working on 2.2-2.6 than on JRuby. So I'll give it a bit more of a whirl, but if I can't make any progress, I'd still like to get this merged in as is, and then maybe try fix the jruby stuff later?

Reply:
Resigning to fixing JRuby later is probably the best course of action at the moment.